src/lib/Instance.ts

```diff
@@ -0,0 +1,24 @@
+export default class Instance {
+  private _onChanged: ((...args: any[]) => void)[];
+  onChanged: {
+    connect: (callback: (...args: any[]) => void) => () => void;
+    fire: (...args: any[]) => void;
+  };
+
+  constructor() {
+    this._onChanged = [];
+    this.onChanged = {
+      connect: (callback: (...args: any[]) => void) => {
+        const length = this._onChanged.length;
+        this._onChanged.push(callback);
+
+        return () => {
+          this._onChanged.splice(length, 1);
+        };
```

Comment:
This disconnect function does not work in the case of removing functions if the order changes; ie

```js
const a = event:Connect(cb1)
const b = event:Connect(cb2)
const c = event:Connect(cb3)
a();
c(); // this wont disconnect anything
```
or
```js
const a = event:Connect(cb1)
const b = event:Connect(cb2)
const c = event:Connect(cb3)
a();
b(); // this will now disconnect c instead of b
```

---

```suggestion
        let active = true;
        const wrappedCallback = (...a:unknown[]) => active ? callback(...a) : undefined;
        this._onChanged.push(callback);

        return () => (active = false);
```